cdb_assist.c

```diff
@@ -338,23 +339,37 @@ static void cdb_gpio(struct cdb_assist *cdb, int gpio, bool on)
 	cdb_ctrl_write(cdb, &cmd[gpio][on], 1);
 }
 
-static void cdb_assist_print_status(struct device *dev)
+static void cdb_assist_print_status(void *data)
+{
+	struct cdb_assist *cdb = data;
+	struct status_value vbat[] = {
+		{
+			.unit = STATUS_MV,
+			.value = cdb->voltage_set,
+		},
+		{
+			.unit = STATUS_MA,
+			.value = cdb->current_actual,
+		},
+		{}
+	};
+	struct status_value vref[] = {
+		{
+			.unit = STATUS_MV,
+			.value = cdb->vref,
+		},
+		{}
+	};
+
+	status_send_values("vbat", vbat);
+	status_send_values("vref", vref);
+}
+
+static void cdb_assist_status_enable(struct device *dev)
```

Comment:
There is no corresponding status_disable function. Maybe it would be easier to leave the server with a single 'send_status' message and keep periodical-request to the client?

Reply:
Then we also need to further complicate the status-cmd, to either allow killing it (status_disable()) or to receive some notification when we want it to send an update.

As proposed we fork the status-cmd and we just consume the data arriving on its stdout, keeping both parts simple.

Reply:
Can we fork a command instead each time we receive a message from the client?
On the other hand if we are measuring power we'd really like to have data at a regular basis. Let's keep your approach then.